### PR TITLE
Firestore init indexes

### PR DESCRIPTION
**Pull request by samtstern**
Finish #562

### CODE REVIEW

File: lib/init/features/firestore.js [hunk 1]
```diff
@@ -108,15 +111,57 @@ var _initIndexes = function(setup, config) {
...
+    .list(projectId)
+    .then(function(indexes) {
+      return JSON.stringify(indexes, null, 2);
+    })
+    .catch(function(e) {
+      if (e.message.indexOf("is not a Cloud Firestore enabled project") >= 0) {
+        return INDEXES_TEMPLATE;
+      }
+
+      throw e;
```
Comment (laurenzlong) [inline lib/init/features/firestore.js]: throw a FirebaseError, see lib/error.js

**Comment by samtstern** [comment]
**Test Case 1 - Empty Folder, Project has custom indexes***

```
samstern@samstern-macbookpro tmp.wkwf5xyH
$ ~/Projects/firebase-tools/bin/firebase init firestore

(emoji redacted)

You're about to initialize a Firebase project in this directory:

  /private/var/folders/q_/sry2gzmj2tb8xsw3s5x_dz4h00643s/T/tmp.wkwf5xyH

Before we get started, keep in mind:

  * You are currently outside your home directory


=== Project Setup

First, let's associate this project directory with a Firebase project.
You can create multiple project aliases by running firebase use --add,
but for now we'll just set up a default project.

? Select a default Firebase project for this directory: FirestoreQuickstarts (firestorequickstarts)

=== Firestore Setup

Firestore Security Rules allow you to define how and when to allow
requests. You can keep these rules in your project directory
and publish them with firebase deploy.

? What file should be used for Firestore Rules? firestore.rules

Firestore indexes allow you to perform complex queries while
maintaining performance that scales with the size of the result
set. You can keep index definitions in your project directory
and publish them with firebase deploy.

? What file should be used for Firestore indexes? firestore.indexes.json

i  Writing configuration info to firebase.json...
i  Writing project information to .firebaserc...

✔  Firebase initialization complete!
samstern@samstern-macbookpro tmp.wkwf5xyH
$ cat firestore.indexes.json
[
  {
    "collectionId": "chat-messages",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "userid",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "timestamp",
        "mode": "DESCENDING"
      }
    ]
  },
  {
    "collectionId": "restaurants",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "category",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "numRatings",
        "mode": "DESCENDING"
      }
    ]
  },
  {
    "collectionId": "restaurants",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "category",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "averageRating",
        "mode": "DESCENDING"
      }
    ]
  },
  {
    "collectionId": "testcollection",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "foo",
        "mode": "DESCENDING"
      },
      {
        "fieldPath": "bar",
        "mode": "ASCENDING"
      }
    ]
  },
  {
    "collectionId": "restaurants",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "city",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "avgRating",
        "mode": "DESCENDING"
      }
    ]
  },
  {
    "collectionId": "restaurants",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "city",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "price",
        "mode": "ASCENDING"
      }
    ]
  },
  {
    "collectionId": "my_collection",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "foo",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "baz",
        "mode": "DESCENDING"
      }
    ]
  },
  {
    "collectionId": "restaurants",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "category",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "city",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "averageRating",
        "mode": "DESCENDING"
      }
    ]
  },
  {
    "collectionId": "restaurants",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "category",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "avgRating",
        "mode": "DESCENDING"
      }
    ]
  },
  {
    "collectionId": "restaurants",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "price",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "numRatings",
        "mode": "DESCENDING"
      }
    ]
  },
  {
    "collectionId": "restaurants",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "price",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "avgRating",
        "mode": "DESCENDING"
      }
    ]
  },
  {
    "collectionId": "restaurants",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "city",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "averageRating",
        "mode": "DESCENDING"
      }
    ]
  },
  {
    "collectionId": "my_collection",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "foo",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "qux",
        "mode": "DESCENDING"
      }
    ]
  },
  {
    "collectionId": "us_cities",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "name",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "state",
        "mode": "ASCENDING"
      }
    ]
  },
  {
    "collectionId": "restaurants",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "category",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "price",
        "mode": "ASCENDING"
      }
    ]
  },
  {
    "collectionId": "foo",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "tags.isGreat",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "time",
        "mode": "ASCENDING"
      }
    ]
  },
  {
    "collectionId": "restaurants",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "category",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "price",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "averageRating",
        "mode": "DESCENDING"
      }
    ]
  },
  {
    "collectionId": "my_other",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "a",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "b",
        "mode": "ASCENDING"
      }
    ]
  },
  {
    "collectionId": "restaurants",
    "state": "READY",
    "fields": [
      {
        "fieldPath": "city",
        "mode": "ASCENDING"
      },
      {
        "fieldPath": "numRatings",
        "mode": "DESCENDING"
      }
    ]
  }
]
```

**Comment by samtstern** [comment]
**Test Case 2 - Empty Folder, Project has no Firestore**

```
$ ~/Projects/firebase-tools/bin/firebase init firestore

     🔥🔥🔥🔥🔥🔥🔥🔥 🔥🔥🔥🔥 🔥🔥🔥🔥🔥🔥🔥🔥  🔥🔥🔥🔥🔥🔥🔥🔥 🔥🔥🔥🔥🔥🔥🔥🔥     🔥🔥🔥     🔥🔥🔥🔥🔥🔥  🔥🔥🔥🔥🔥🔥🔥🔥
     🔥🔥        🔥🔥  🔥🔥     🔥🔥 🔥🔥       🔥🔥     🔥🔥  🔥🔥   🔥🔥  🔥🔥       🔥🔥
     🔥🔥🔥🔥🔥🔥    🔥🔥  🔥🔥🔥🔥🔥🔥🔥🔥  🔥🔥🔥🔥🔥🔥   🔥🔥🔥🔥🔥🔥🔥🔥  🔥🔥🔥🔥🔥🔥🔥🔥🔥  🔥🔥🔥🔥🔥🔥  🔥🔥🔥🔥🔥🔥
     🔥🔥        🔥🔥  🔥🔥    🔥🔥  🔥🔥       🔥🔥     🔥🔥 🔥🔥     🔥🔥       🔥🔥 🔥🔥
     🔥🔥       🔥🔥🔥🔥 🔥🔥     🔥🔥 🔥🔥🔥🔥🔥🔥🔥🔥 🔥🔥🔥🔥🔥🔥🔥🔥  🔥🔥     🔥🔥  🔥🔥🔥🔥🔥🔥  🔥🔥🔥🔥🔥🔥🔥🔥

You're about to initialize a Firebase project in this directory:

  /private/var/folders/q_/sry2gzmj2tb8xsw3s5x_dz4h00643s/T/tmp.wkwf5xyH

Before we get started, keep in mind:

  * You are currently outside your home directory


=== Project Setup

First, let's associate this project directory with a Firebase project.
You can create multiple project aliases by running firebase use --add,
but for now we'll just set up a default project.

? Select a default Firebase project for this directory: FirePerfDevFeedback (fireperfdevfeedback)

=== Firestore Setup

Firestore Security Rules allow you to define how and when to allow
requests. You can keep these rules in your project directory
and publish them with firebase deploy.

? What file should be used for Firestore Rules? firestore.rules

Firestore indexes allow you to perform complex queries while
maintaining performance that scales with the size of the result
set. You can keep index definitions in your project directory
and publish them with firebase deploy.

? What file should be used for Firestore indexes? firestore.indexes.json

i  Writing configuration info to firebase.json...
i  Writing project information to .firebaserc...

✔  Firebase initialization complete!
samstern@samstern-macbookpro tmp.wkwf5xyH
$ head firestore.indexes.json
{
  // Example:
  //
  // "indexes": [
  //   {
  //     "collectionId": "widgets",
  //     "fields": [
  //       { "fieldPath": "foo", "mode": "ASCENDING" },
  //       { "fieldPath": "bar", "mode": "DESCENDING" }
  //     ]
```

**Comment by samtstern** [comment]
**Test Case 3 - Existing file, project has custom indexes**
```
$ head firestore.indexes.json
{ "indexes": [] }
samstern@samstern-macbookpro tmp.wkwf5xyH
$ ~/Projects/firebase-tools/bin/firebase init firestore

(emoji redacted) 

You're about to initialize a Firebase project in this directory:

  /private/var/folders/q_/sry2gzmj2tb8xsw3s5x_dz4h00643s/T/tmp.wkwf5xyH

Before we get started, keep in mind:

  * You are currently outside your home directory


=== Project Setup

First, let's associate this project directory with a Firebase project.
You can create multiple project aliases by running firebase use --add,
but for now we'll just set up a default project.

? Select a default Firebase project for this directory: FirestoreQuickstarts (firestorequickstarts)

=== Firestore Setup

Firestore Security Rules allow you to define how and when to allow
requests. You can keep these rules in your project directory
and publish them with firebase deploy.

? What file should be used for Firestore Rules? firestore.rules

Firestore indexes allow you to perform complex queries while
maintaining performance that scales with the size of the result
set. You can keep index definitions in your project directory
and publish them with firebase deploy.

? What file should be used for Firestore indexes? firestore.indexes.json
? File firestore.indexes.json already exists. Do you want to overwrite it with the Firestore Indexes from the Fireba
se Console? Yes

i  Writing configuration info to firebase.json...
i  Writing project information to .firebaserc...

✔  Firebase initialization complete!
samstern@samstern-macbookpro tmp.wkwf5xyH
$ head firestore.indexes.json
{
  "indexes": [
    {
      "collectionId": "chat-messages",
      "state": "READY",
      "fields": [
        {
          "fieldPath": "userid",
          "mode": "ASCENDING"
        },
```

**Comment by laurenzlong** [comment]
Thanks Sam!

**Pull request merged**